fedimint-api/src/module/mod.rs

```diff
@@ -353,6 +359,12 @@ pub trait ModuleGen: Debug + Sized {
     fn validate_config(&self, identity: &PeerId, config: ServerModuleConfig) -> anyhow::Result<()>;
 
     fn hash_client_module(&self, config: serde_json::Value) -> anyhow::Result<sha256::Hash>;
+
+    async fn dump_module_database(
```

Comment:
This shouldn't be on `ModuleGen`, but on `Module`, no? We want to dump things from a specific module instance.

Reply:
Maybe add `asyn fn list_db_prefix_keys() -> Vec<String>` so the user can just list these.

Reply:
`_module_` here is redundant, IMO.

Reply:
> This shouldn't be on `ModuleGen`, but on `Module`, no? We want to dump things from a specific module instance.

Hmm do we need to? I think if we want to move it to `Module`, we'd need to instantiate each module (`FedimintConsensus::new`), which for some modules, like the wallet module, instantiates new threads and does other initialization. In this case we just want to read the database, we don't actually need to instantiate the module. The current code does dump things from a specific module instance, it just grabs it from the server config.

I don't have a strong opinion but I will change it if you feel its a better design.

Reply:
That's true. But then - shouldn't we pass the instance id here, so the implementation knows *which* module to dump (there might be multiple of the same kind). Or is this already handled by `dbtx` being isolated?

Reply:
Yep, this is handled by `dbtx` since it is isolated. When the returned data is added to the BTreeMap that is eventually serialized as JSON, it is added as "<kind>-<module instance id>" so when looking at the dump output it is easy to tell the difference (e.g. Mint-2)